tests/utils/test_db.py

```diff
@@ -0,0 +1,124 @@
+import re
+from unittest.mock import MagicMock, patch
+
+from page_analyzer.utils.db import (
+    create_check,
+    create_url,
+    get_url_by_name,
+    get_url_checks,
+    get_url_data,
+    get_urls,
+)
+
+
+def test_create_url(test_database: None):
+    # given
+    url_name = "https://example.com"
+
+    # when
+    url_id = create_url(url_name)
+
+    # then
+    created_url = get_url_data(url_id)
+    assert created_url["name"] == url_name
+    created_url_id = get_url_by_name(url_name)
+    assert url_id == created_url_id
+
+
+def test_get_url_data_none(test_database: None):
+    # given
+    url_id = 1
+
+    # when
+    url_data_result = get_url_data(url_id)
+
+    # then
+    assert url_data_result is None
+
+
+def test_get_url_by_name_none(test_database: None):
+    # given
+    url_name = "https://example.com"
+
+    # when
+    url_id = get_url_by_name(url_name)
+
+    # then
+    assert url_id is None
+
+
+def test_get_urls(test_database: None):
+    create_url("https://example1.com")
+    create_url("https://example2.com")
+
+    # when
+    urls_data = get_urls()
+
+    # then
+    match_count = 0
+    for url_data in urls_data:
+        assert re.search(r"https://example\d.com", url_data["url_name"])
+        match_count += 1
+    assert match_count == 2
+
+
+def test_get_urls_empty(test_database: None):
+    # when
+    urls_data = get_urls()
+
+    # then
+    assert len(urls_data) == 0
+
+
+def test_create_check(test_database: None):
+    url_id = create_url("https://example1.com")
+    url_data = get_url_data(url_id)
+
+    request_response_mock = MagicMock()
+    request_response_mock.return_value = {
+        "response_code": "200",
+        "h1": "some header",
+        "title": "some title",
+        "description": "some description",
+    }
+
+    # when
+    with patch(
+        "page_analyzer.utils.db.check_url",
+        return_value=request_response_mock.return_value,
+    ) as mock_requests:
```

Comment:
you can use `return_value` right away

```suggestion

    # when
    with patch(
        "page_analyzer.utils.db.check_url",
        return_value={
            "response_code": "200",
            "h1": "some header",
            "title": "some title",
            "description": "some description",
        },
    ) as mock_check_url:
```